packages/okta-angular/src/okta/okta.module.ts

```diff
@@ -16,26 +16,39 @@ import { OktaCallbackComponent, OktaLoginRedirectComponent } from './components/
 import { OktaAuthService } from './services/okta.service';
 import { OktaAuthGuard } from './okta.guard';
 import { OktaConfig, OKTA_CONFIG } from './models/okta.config';
+import { createOktaService } from './createService';
+import { Router } from '@angular/router';
 
 @NgModule({
   declarations: [
     OktaCallbackComponent,
-    OktaLoginRedirectComponent
+    OktaLoginRedirectComponent,
   ],
   exports: [
     OktaCallbackComponent,
-    OktaLoginRedirectComponent
+    OktaLoginRedirectComponent,
+  ],
+  providers: [
+    OktaAuthGuard,
+    {
+      provide: OktaAuthService,
+      useFactory: createOktaService,
+      deps: [
+        OKTA_CONFIG,
+        Router
+      ]
+    }
   ]
 })
 export class OktaAuthModule {
+  // Deprecated. Your app should provide OKTA_CONFIG directly
```

Comment:
Do we have any kind of logger that we can spit out a deprecation message to if this is used?

Reply:
I don't think we have one. We could use console.error or the debug() npm module. I decided to leave that out since i thought it would just create another issue for us. (how do i remove this message?) I'm not sure how many customers are using AOT.